### PR TITLE
#3087/#3099: Quick Content Fixes - [MS]

### LINKED ISSUE #3087: Error messages for zip code and email address differs between org model and non-org model instances
### Issue description

Recently error messages were changed for zip codes and email address for org-model.  We need to make sure the changes are also made to non org model instances.

If it makes sense, also look into centralizing the error messages.

### Acceptance criteria
These should already be done for org model and needs to be done for non-org model cases
- [x] On the "Organization" page:
  - [x] When zip code is required and left blank, update error message to read: 
    - Enter a 5-digit or 9-digit zip code, like 12345 or 12345-6789.
  - [x] When zip code uses wrong format, update the error message to match the message listed above. (we want them to be consistent)
- [x] On the "add a domain manager" page:
  - [x] When email address is required and left blank, update error message to read: 
    - Enter an email address in the required format, like name@example.com.
  - [x] The error message should appear at the top of the page and at the field level. Currently, it's only displayed at the field level.
    - This should be updated for org model AND non-org model
- [x] Review registrar and verify other instances where these fields appear show this updated messaging

### Additional context
See parent ticket, especially [comment from the parent ticket](https://github.com/cisagov/manage.get.gov/issues/2777#issuecomment-2471296151)

### Links to other issues

_No response_

### LINKED ISSUE #3099: Remove duplicate sentence from Senior Official page (org and non-org model)
### Issue description

When a senior official hasn't been assigned to a domain or organization, unique messaging displays in the UI. 

Currently, there's a duplicate sentence displaying in those scenarios. 

### Acceptance criteria

- [ ] **Non-org model:** Remove duplicate sentence from senior official page when an SO hasn't been assigned.
- [ ] **Org model:** Remove duplicate sentence from senior official page when an SO hasn't been assigned.

### **Non-org model**
Image

### **Org model**
Image

### Links to other issues

_No response_

### PR DESCRIPTION
**Pull request by Matt-Spence**
## Ticket

Resolves #3087 
Resolves #3099 

## Changes

- Ensure error messages display in messages when adding domain managers
- remove duplicate sentences from senior official page when no senior official is found

## Code Review Verification Steps
Put in an invalid email address when adding a domain manager -> check that the correct error message displays (including at the top of the page and on the input itself)

View senior officials in org and non-org mode for domains that don't have senior officials listed. Check that duplicate sentences have been removed.

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Update documentation in READMEs and/or onboarding guide

#### Ensured code standards are met (Original Developer)

- [ ] If any updated dependencies on Pipfile, also update dependencies in requirements.txt.
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] Tag @dotgov-designers in this PR's Reviewers for design review. If code is not user-facing, delete design reviewer checklist
- [ ] Verify new pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Verified code meets all checks above. Address any checks that are not satisfied
- [ ] Reviewed this code and left comments. Indicate if comments must be addressed before code is merged
- [ ] Checked that all code is adequately covered by tests
- [ ] Verify migrations are valid and do not conflict with existing migrations

#### Validated user-facing changes as a developer
**Note:** Multiple code reviewers can share the checklists above, a second reviewer should not make a duplicate checklist. All checks should be checked before approving, even those labeled N/A. 

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] (Rarely needed) Tested as both an analyst and applicant user

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior. Comment any found issues or broken flows.
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### References
- [Code review best practices](../docs/dev-practices/code_review.md)

### CODE REVIEW

File: src/registrar/templates/domain_add_user.html [hunk 1]
```diff
@@ -37,6 +37,9 @@
     </nav>
     {% endif %}
   {% endblock breadcrumb %}
+
+  {% include "includes/form_errors.html" with form=form %}
+
```
Comment (asaki222) [inline src/registrar/templates/domain_add_user.html:42]: I read on the ticket that the error is supposed to appear on the field level and at the top of the page. Was there an update that wasn't noted? I don't get the error at the top

Screenshot 2024-12-23 at 10 24 24 AM

Comment (Matt-Spence) [reply src/registrar/templates/domain_add_user.html:42]: This wasn't deployed to MS at the time you did the review, sorry. I have two PR's up but forgot that I only have one sandbox 🫠. Should deploy shortly.

**Review by asaki222 — APPROVED** [review]
LGTM!

**Review by kristinacyin — CHANGES_REQUESTED** [review]
A couple of things I noticed: 

1. **Multiple success messages:** There's two success alerts after I correct an erroneous zip code in the Organization page (org model). Noticed that same thing happens for other fields as well on the page.
image

2. **Placement of page error alert**: 

Usually page errors are placed above the heading like so: 
image

However, the page error alerts (for all fields) on the Organization page are below the heading, under the instructional paragraph. Was this intentional? Can we move them above so that they're more consistent?
getgov-ms app cloud gov_organization_

**Comment by Matt-Spence** [comment]
> A couple of things I noticed:
> 
> 1. **Multiple success messages:** There's two success alerts after I correct an erroneous zip code in the Organization page (org model). Noticed that same thing happens for other fields as well on the page.
>    image
> 2. **Placement of page error alert**:
> 
> Usually page errors are placed above the heading like so: image
> 
> However, the page error alerts (for all fields) on the Organization page are below the heading, under the instructional paragraph. Was this intentional? Can we move them above so that they're more consistent? getgov-ms app cloud gov_organization_

I think these are all on main 🫠 I didn't change anything on the organization or security email pages!

**Review by kristinacyin — APPROVED** [review]
LGTM! Thank you for the additional fix!

**Pull request merged**